cmake/CMakeAM.cmake

```diff
@@ -125,13 +131,74 @@ function(convert_automake_internal AM_INPUT_FILE CMAKE_OUTPUT_FILE)
   endforeach()
 endfunction()
 
+function(am_find_directories INPUT_PATH OUTPUT_LIST)
+  if ("${INPUT_PATH}" MATCHES "/")
+    string(REGEX REPLACE "/[^/]*$" "" INPUT_DIRECTORY "${INPUT_PATH}")
+    list(APPEND "${OUTPUT_LIST}" "${INPUT_DIRECTORY}")
+  endif()
+
+  set(${OUTPUT_LIST} "${${OUTPUT_LIST}}" PARENT_SCOPE)
+endfunction()
+
+# Set target CXXFLAGS, by parsing for special values
+function(am_set_flags TARGET FLAGS_LIST)
+  list(LENGTH FLAGS_LIST FLAGS_COUNT)
+
+  set(FLAGS_MISC)
+  set(FLAGS_DEFINE)
+  set(FLAGS_INCLUDES)
+  set(FLAGS_INCLUDES_SYSTEM)
+
+  set(i 0)
+  while(${i} LESS ${FLAGS_COUNT})
+    list(GET FLAGS_LIST ${i} FLAG)
+
+    if("${FLAG}" MATCHES "^-D$")
+      math(EXPR i "${i} + 1")
+      list(GET FLAGS_LIST ${i} FLAG)
+      list(APPEND FLAGS_DEFINE "${FLAG}")
+    elseif("${FLAG}" MATCHES "^-D")
+      string(REGEX REPLACE "^-D" "" FLAG "${FLAG}")
+      list(APPEND FLAGS_DEFINE "${FLAG}")
+    elseif("${FLAG}" STREQUAL "-isystem")
+      math(EXPR i "${i} + 1")
+      list(GET FLAGS_LIST ${i} FLAG)
+      list(APPEND FLAGS_INCLUDES_SYSTEM "${FLAG}")
+    elseif("${FLAG}" MATCHES "^-I$")
+      math(EXPR i "${i} + 1")
+      list(GET FLAGS_LIST ${i} FLAG)
+      list(APPEND FLAGS_INCLUDES "${FLAG}")
+    elseif("${FLAG}" MATCHES "^-I")
+      string(REGEX REPLACE "^-I" "" FLAG "${FLAG}")
+      list(APPEND FLAGS_INCLUDES "${FLAG}")
+    else()
+      list(APPEND FLAGS_MISC "${FLAG}")
+    endif()
+
+    math(EXPR i "${i} + 1")
+  endwhile()
+
+  string(REPLACE ";" " " FLAGS_MISC "${FLAGS_MISC}")
+  set_target_properties(${TARGET} PROPERTIES APPEND_STRING PROPERTY COMPILE_FLAGS "${FLAGS_MISC}")
```

Comment:
Yeah, interestingly enough it does work. Or at least it does for me, when I tested it.

I found it recommended in a StackOverflow question here: http://stackoverflow.com/a/19161129 , that's how I got that idea. :P

Is there a difference between `set_target_properties(...)` and `set_property(TARGET ...)`? Should I rather use one over the other?